PluginClasses/ServiceApi.swift

```diff
@@ -0,0 +1,93 @@
+//
+//  ServiceApi.swift
+//  TVADataBinder
+//
+//  Created by MSApps on 16/04/2020.
+//
+
+import Foundation
+
+let tokenKey = "AWSCognitoTokenKey"
+let defaults = UserDefaults.standard
+var baseApi :String?
+let favouritePath = "/userlists/favorites/"
+
+func getFavouriteState(uid: String, completion: @escaping (( _ on: Bool?) -> Void)){
+    guard let token = defaults.string(forKey: tokenKey), let baseApiUrl = baseApi else{
+        return
+    }
+    let apiName = "\(baseApiUrl)\(favouritePath)\(uid)"
+    let url = URL(string: apiName)!
+    let request = NSMutableURLRequest(url: url)
+    request.setValue("Bearer \(token)", forHTTPHeaderField: "Authorization")
+    request.httpMethod = "GET"
+    request.setValue("application/json", forHTTPHeaderField: "Content-Type")
+    makeRequest(request: request) { (responseJson, response, error) in
+        if(response?.statusCode == 200){
+            completion(true)
+        }else if (response?.statusCode == 404){
```

Comment:
what about not modified?

https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/304

Reply:
Explained on slack